### PR TITLE
Random events

### PR DESCRIPTION
**Pull request by LoganTHarvell**
Added initial random event mechanics for planets

Includes:
- plague (high death rate)
- blight (reduced farming)
- mine collapses (reduced mining)

@iholdeman1 this branch can be pulled from to update Events UI with random events

### CODE REVIEW

File: Astral Pathfinder/Game/PlanetManager.cpp [hunk 1]
```diff
@@ -237,7 +236,7 @@ void PlanetManager::handleCollisions(ShipManager *sm, Uint32 frame) {
   // If player has docked checks docked planet for continued collision
   else {
     ColliderComponent collider = planets[dockedPlanetIndex].getCollider();
-    if (!collider.collisionOBB(playerVertices, playerAngle)) {
```
Comment (holdemani) [inline Astral Pathfinder/Game/PlanetManager.cpp:240]: What happened to playerAngle?

File: Astral Pathfinder/Models/Planet.cpp [hunk 2]
```diff
@@ -174,6 +176,22 @@ void Planet::updateStatus() {
...
+  // Exits immediately if planet has no population
+  if (population <= 0) return;
+  
+  // Updates random event flags for growth period
+  if ((frame-frameDocked)%growthPeriod == 0) {
+    float randMax = static_cast<float>(RAND_MAX);
+
+    events.plague = (plagueRate > rand()/randMax);
+    events.blight = (blightRate > rand()/randMax);
+    events.mineCollapse = (mineCollapseRate > rand()/randMax);
```
Comment (holdemani) [inline Astral Pathfinder/Models/Planet.cpp:191]: This means that a planet can have multiple events happening at once right?

**Review by holdemani — COMMENTED** [review]
See comments

File: Astral Pathfinder/Game/PlanetManager.cpp (hunk 1 above)
Comment (LoganTHarvell) [reply Astral Pathfinder/Game/PlanetManager.cpp:240]: You don’t need player angle in the collision detection.

File: Astral Pathfinder/Models/Planet.cpp (hunk 2 above)
Comment (LoganTHarvell) [reply Astral Pathfinder/Models/Planet.cpp:191]: Yes

**Pull request merged**